geps/gep-1713/index.md

```diff
@@ -0,0 +1,405 @@
+2. ListenerSet ordered by creation time (oldest first)
+3. ListenerSet ordered alphabetically by “{namespace}/{name}”.
+
+If there are listener conflicts, this should be reported as `Conflicted=True` in the ListenerStatus as usual. See 'Conditions' section below for more details on object should report the conflict.
+
+###  Gateway Conditions
+
+`Gateway` currently supports the following top-level condition types: `Accepted` and `Programmed`
+
+For a `Gateway`, `Accepted` should be set based on the entire set of merged listeners.
+For instance, if a `ListenerSet` listener is invalid, `ListenersNotValid` would be reported. 
+`Programmed` is not expected, generally, to depend on the children resources, but if an implementation does depend on these
+they should consider child resources when reporting this status.
+
+Parent gateways MUST NOT have `ListenerSet` listeners in their `status.listeners` conditions list.
+
+It is up to the implementation whether an invalid listener affects other listeners in the Gateway.
+
+### ListenerSet Conditions
+
+`ListenerSets` MUST NOT have their parent `Gateway`'s' listeners in the `status.listeners` conditions list.  An implementation MAY reject listeners with `ListenerConditionAccepted=False` and Reason `TooManyListeners` `ListenerSets`, like a `Gateway`, also have two top-level conditions: `Accepted` and `Programmed`. These conditions, when surfacing details about listeners, MUST only summarize the `status.listener` conditions that are exclusive to the `ListenerSet`. 
+
+These conditions MUST also surface top-level `Gateway` conditions that impact the `ListenerSet`. For example, if a `Gateway` requests an invalid address and it cannot be accepted/programmed then the `ListenerSet`'s' `Accepted` condition MUST be set to `False`.
+
+For example, if I have a `Gateway` named `parent`, and two `ListenerSets` named `child-1`, and `child-2` then:
+* If `parent` is entirely invalid (for example, an invalid `address`) and `Accepted=False`, all two `ListenerSets` will reported `Accepted=False`.
+* If `child-1` has an invalid listener, `parent` and `child-1` will report `ListenersNotValid`, while `child-2` will not.
+* If `child-1` references a parent that doesn't allow merging then `child-1` will report `Accepted=False`
+* If `child-1` references another child (eg. `child-2`) then `child-1` will report `Accepted=False` 
+* If `child-1` is valid, then when `child-2` is created if it conflicts with `child-1` then `child-2` will report `Accepted=False`. `child-1` status conditions will remain unchanged. `parent` will report `ListenersNotValid`
+
+When reporting status of a child, an implementation SHOULD be cautious about what information from the parent or siblings are reported
+to avoid accidentally leaking sensitive information that the child would not otherwise have access to.
+
+#### Policy Attachment
+
+Policies attached to a parent `Gateway` apply to both the parent and all `ListenerSet` listeners.
```

Comment:
Conceptually, does that mean such kinds of policy are always of the _Inherited_ class?

What would be the recommendation for defining a policy that is NOT supposed to apply to a particular `ListenerSet`? Perhaps split the Gateway into more objects and target at the lower levels?

Reply:
For context: I added this as a starting point for guidance and I'm flexible on what seems the best here. I'm not an expert on PolicyAttachment so input is appreciated.

> Conceptually, does that mean such kinds of policy are always of the Inherited class?

I believe so - for example a Policy on the Gateway in my mind applies to all the listeners and thus is inherited by the ListenerSet resource.

> What would be the recommendation for defining a policy that is NOT supposed to apply to a particular ListenerSet? Perhaps split the Gateway into more objects and target at the lower levels?

This is a good question and I'm curious if there's any existing precedent. For example given a Gateway with N listeners is it currently possible to exclude some listeners M from a policy?

To answer your question - I think with ListenerSet then it could be possible to do the following:

1. Have no policy apply to the Gateway - but don't have any listeners.
2. Apply different policies on `ListenerSets`

Reply:
I simplified the language here to say policy _can_ be inherited from the Gateway.

Also highlighted that applying policy to a listenerset allows different sets that are part of the gateway to have different policies.